server/src/main/java/org/elasticsearch/health/HealthPeriodicLogger.java

```diff
@@ -254,6 +323,31 @@ public void onFailure(Exception e) {
         }
     };
 
+    /**
+     * Handle the setting (and possible creation) of APM metrics
+     */
+    // default visibility for testing purposes
+    void handleMetrics(List<HealthIndicatorResult> healthIndicatorResults) {
+        if (healthIndicatorResults != null) {
+            for (HealthIndicatorResult result : healthIndicatorResults) {
+                String metricName = result.name();
+                LongGaugeMetric metric = this.metrics.get(metricName);
+                if (metric == null) {
+                    metric = LongGaugeMetric.create(
+                        this.meterRegistry,
+                        String.format(Locale.ROOT, "es.health.%s.red", metricName),
+                        String.format(Locale.ROOT, "%s: Red", metricName),
+                        "{cluster}"
+                    );
+                    this.metrics.put(metricName, metric);
+                }
+                metricWriter.accept(metric, (long) (result.status() == RED ? 1 : 0));
```

Comment:
I think we can get rid of the cast with:

```suggestion
                metricWriter.accept(metric, (result.status() == RED ? 1L : 0L));
```